packages/wkt/test/wkt.test.js

```diff
@@ -604,3 +604,404 @@ test('should parse a MULTIPOLYGON with a hole', function (t) {
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: [40, 10]
+      },
+      {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [40, 40],
+            [20, 45],
+            [45, 30],
+            [40, 40]
+          ]
+        ]
+      }
+    ]
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION without polygons', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40))';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: [40, 10]
+      },
+      {
+        type: 'LineString',
+        coordinates: [
+          [10, 10],
+          [20, 20],
+          [10, 40]
+        ]
+      }
+    ]
+  });
+});
+
+test('should parse a empty GEOMETRYCOLLECTION', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION EMPTY';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: []
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with empty geometries', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION ( POINT EMPTY )';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: []
+      }
+    ]
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with a empty geometry and a non empty geometry', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION ( POINT EMPTY, POINT (40 10))';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: []
+      },
+      {
+        type: 'Point',
+        coordinates: [40, 10]
+      }
+    ]
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with Z property', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION Z ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40)))';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: [40, 10]
+      },
+      {
+        type: 'LineString',
+        coordinates: [
+          [10, 10],
+          [20, 20],
+          [10, 40]
+        ]
+      },
+      {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [40, 40],
+            [20, 45],
+            [45, 30],
+            [40, 40]
+          ]
+        ]
+      }
+    ],
+    properties: {
+      z: true
+    }
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with M property', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION M ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40)))';
```

Comment:
you're calling this a geometry collection with M values, but none of the coordinates have a third value...

Reply:
I struggled to find a standard for this scenario. It was unclear if Specifying M or Z on GeometryCollection directly forces it's children to have a third attribute. You probably know the correct resolution here. It seems entirely probable that a collection could support both 2D, 3D and 4D attributes. To me the Z and M feel like noop values.

Reply:
I think it's fair to assume that if geometry collection has one of the attributes, then all of it's children likely would as well. I'll go forward with that assumption and fix the tests.

Reply:
I hope these changes reflect what I intended above